packages/eslint-plugin-tslint/src/index.ts

```diff
@@ -0,0 +1,168 @@
+import { Rule } from 'eslint';
+import memoize from 'lodash.memoize';
+import { Configuration, RuleSeverity } from 'tslint';
+import { Program } from 'typescript';
+import { CustomLinter } from './custom-linter';
+
+//------------------------------------------------------------------------------
+// Plugin Definition
+//------------------------------------------------------------------------------
+
+/**
+ * @todo share types between packages
+ */
+interface ParserServices {
+  program: Program | undefined;
+  esTreeNodeToTSNodeMap: WeakMap<object, any> | undefined;
+  tsNodeToESTreeNodeMap: WeakMap<object, any> | undefined;
+}
+
+type RawRuleConfig =
+  | null
+  | undefined
+  | boolean
+  | any[]
+  | {
+      severity?: RuleSeverity | 'warn' | 'none' | 'default';
+      options?: any;
+    };
+
+interface RawRulesConfig {
+  [key: string]: RawRuleConfig;
+}
+
+/**
+ * Construct a configFile for TSLint
+ */
+const tslintConfig = memoize(
+  (
+    lintFile: string,
+    tslintRules: RawRulesConfig,
+    tslintRulesDirectory: string[]
+  ) => {
+    if (lintFile != null) {
+      return Configuration.loadConfigurationFromPath(lintFile);
+    }
+    return Configuration.parseConfigFile({
+      rules: tslintRules || {},
+      rulesDirectory: tslintRulesDirectory || []
+    });
+  },
+  (lintFile: string | undefined, tslintRules = {}, tslintRulesDirectory = []) =>
+    `${lintFile}_${Object.keys(tslintRules).join(',')}_${
+      tslintRulesDirectory.length
+    }`
+);
+
+export const rules = {
+  /**
+   * Expose a single rule called "config", which will be accessed in the user's eslint config files
+   * via "tslint/config"
+   */
+  config: {
+    meta: {
+      docs: {
+        description:
+          'Wraps a TSLint configuration and lints the whole source using TSLint',
+        category: 'TSLint'
+      },
+      schema: [
+        {
+          type: 'object',
+          properties: {
+            rules: {
+              type: 'object',
+              /**
+               * No fixed schema properties for rules, as this would be a permanently moving target
+               */
+              additionalProperties: true
+            },
+            rulesDirectory: {
+              type: 'array',
+              items: {
+                type: 'string'
+              }
+            },
+            lintFile: {
+              type: 'string'
+            }
+          },
+          additionalProperties: false
+        }
+      ]
+    },
+    create: function(context: Rule.RuleContext) {
+      const fileName = context.getFilename();
+      const sourceCode = context.getSourceCode().text;
+      const parserServices: ParserServices | undefined = context.parserServices;
```

Comment:
this will make typescript program reused by both `eslint-plugin-typescript` and `eslint-plugin-tslint`

it will improve performance while running both of plugins at same time. 👍